spec/lib/tasks/update_schemes_and_locations_from_csv_spec.rb

```diff
@@ -0,0 +1,454 @@
+      let!(:closed_collection_lettings_log) { FactoryBot.create(:lettings_log, :sh, scheme: schemes[0], location:, values_updated_at: nil, owning_organisation: schemes[0].owning_organisation) }
+      let!(:archived_closed_collection_lettings_log) { FactoryBot.create(:lettings_log, :sh, scheme: schemes[0], location:, values_updated_at: nil, owning_organisation: schemes[0].owning_organisation) }
+
+      before do
+        closed_collection_lettings_log.startdate = Time.zone.local(2022, 4, 1)
+        closed_collection_lettings_log.save!(validate: false)
+        archived_closed_collection_lettings_log.startdate = Time.zone.local(2021, 4, 1)
+        archived_closed_collection_lettings_log.save!(validate: false)
+        allow(storage_service).to receive(:get_file_io)
+        .with("original_schemes.csv")
+        .and_return(StringIO.new(replace_entity_ids(schemes[0], schemes[1], schemes[2], File.open("./spec/fixtures/files/original_schemes.csv").read)))
+
+        allow(storage_service).to receive(:get_file_io)
+        .with("updated_schemes.csv")
+        .and_return(StringIO.new(replace_entity_ids(schemes[0], schemes[1], schemes[2], File.open("./spec/fixtures/files/updated_schemes.csv").read)))
+      end
+
+      it "updates the allowed scheme fields if they have changed and doesn't update other fields" do
+        create(:organisation_relationship, parent_organisation: schemes[0].owning_organisation, child_organisation: different_organisation)
+        closed_collection_lettings_log.update!(startdate: Time.zone.now)
+        archived_closed_collection_lettings_log.update!(startdate: Time.zone.now)
+
+        task.invoke(original_schemes_csv_path, updated_schemes_csv_path)
+        schemes[0].reload
+        expect(schemes[0].service_name).to eq("Updated test name")
+        expect(schemes[0].sensitive).to eq("No")
+        expect(schemes[0].registered_under_care_act).to eq("No")
+        expect(schemes[0].support_type).to eq("Low level")
+        expect(schemes[0].scheme_type).to eq("Direct Access Hostel")
+        expect(schemes[0].arrangement_type).to eq("Another registered stock owner")
+        expect(schemes[0].intended_stay).to eq("Permanent")
+        expect(schemes[0].primary_client_group).to eq("People with drug problems")
+        expect(schemes[0].has_other_client_group).to eq("No")
+        expect(schemes[0].owning_organisation).to eq(different_organisation)
+        expect(schemes[0].created_at).to eq(Time.zone.local(2021, 4, 1))
+        expect(schemes[0].total_units).to eq(2)
+      end
+
+      it "updates the lettings log if scheme has changed owning organisation" do
+        create(:organisation_relationship, parent_organisation: schemes[0].owning_organisation, child_organisation: different_organisation)
+        closed_collection_lettings_log.update!(startdate: Time.zone.now)
```

Comment:
Seems a bit annoying to have to do this - overall I think it makes sense but I had to think about why. If there's an obvious way to avoid this I'd be tempted but I don't think it's that important.

Reply:
Yeah, you're right, it's confusing. I renamed and flipped the dates we set so it's only set to old dates in relevant tests